test/functional/users_controller_test.rb

```diff
@@ -201,6 +201,22 @@ def setup
     assert_equal User.find(user.id).bio, 'Hello, there!'
   end
 
+  test 'username should not be updated1' do
+    user = users(:bob)
+    UserSession.create(user)
+    user.username = 'newval'
+    user.save
+    assert_equal user.username, 'newval'
```

Comment:
Try this
`u = users(:bob)
user = users(:bob)
    UserSession.create(user)
    user.username = 'newval'
    user.save
    assert_equal u.username, 'newval' `
I think then we can debug it.